### PR TITLE
Feature/SOF-6690: generate & publish docs

### PR DESCRIPTION
**Pull request by mrakitin**
A follow-up PR based on #74 to publish the documentation to https://exabyte-io.github.io/api-examples/.

Uses:
- https://www.mkdocs.org
- https://oprypin.github.io/mkdocs-same-dir
- https://github.com/apenwarr/mkdocs-exclude
- https://github.com/danielfrg/mkdocs-jupyter
- https://squidfunk.github.io/mkdocs-material/

### CODE REVIEW

**Comment by mrakitin** [comment]
https://github.com/Exabyte-io/api-examples/pull/74 should be merged first.

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+      - name: Run ruff linter
+        uses: ./actions/py/lint
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run pre-commit hooks
+        run: |
+          python -m pip install pre-commit
+          pre-commit run --all
```
Comment (chu-k) [inline .github/workflows/cicd.yml:37]: Ruff is included in the pre-commit hooks. Perhaps we can change the "lint" action here to run all the pre-commit hooks instead. https://github.com/Exabyte-io/actions/blob/main/py/lint/action.yml

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  run-tests:
```
Comment (chu-k) [inline .github/workflows/cicd.yml:39]: job name could maybe be "build-and-deploy-docs" to be more accurate

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (mrakitin) [reply .github/workflows/cicd.yml:37]: Correct, I was thinking between extending the existing `./actions/py/lint` and adding the checks here. With pre-commit, we can specify all but the `ruff` check, but I'd rather do it consistently via the pre-commit.

File: .github/workflows/cicd.yml (hunk 2 above)
Comment (mrakitin) [reply .github/workflows/cicd.yml:39]: Right, copy-paste leftover 🙂

File: README.md [hunk 3]
```diff
@@ -50,8 +50,7 @@ Follow the steps below in order to setup and view the Jupyter notebooks:
     cd api-examples
     virtualenv .env
     source .env/bin/activate
-    pip install --no-deps -r requirements.txt
-    pip install --no-deps -e .
+    pip install -e ."[jupyter]"
```
Comment (timurbazhirov) [inline README.md:53]: As discussed, let's split the environments to:
- colab
- localhost
- docs

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**